src/support/string.h

```diff
@@ -75,7 +75,14 @@ inline bool isNumber(const std::string& str) {
   return !str.empty() && std::all_of(str.begin(), str.end(), ::isdigit);
 }
 
-std::ostream& printEscaped(std::ostream& os, std::string_view str);
+enum EscapeMode {
+  Normal,
+  JSON
+};
+
+std::ostream& printEscaped(std::ostream& os,
```

Comment:
I think having a separate `printJSONEscaped` function would make more sense for this interface, even if the implementation still uses a flag.